main.go

```diff
@@ -976,9 +1022,12 @@ func transformLink(args []string) ([]string, error) {
 			pkgPath = buildInfo.firstImport
 		}
 		if id := buildInfo.imports[pkgPath].buildID; id != "" {
+			newName, ok := nameMap[pkg+"."+name]
+			if !ok {
+				newName = hashWith(id, name)
```

Comment:
document this if/else; why do we sometimes use nameMap, and sometimes hashWith? this is not obvious at all from just reading the code.